react/features/toolbar/actions.web.js

```diff
@@ -0,0 +1,291 @@
+ */
+export function showDesktopSharingButton(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonName = 'desktop';
+        const isVisible
+            = APP.conference.isDesktopSharingEnabled
+                && UIUtil.isButtonEnabled(buttonName);
+
+        dispatch(setToolbarButton(buttonName, {
+            hidden: !isVisible
+        }));
+    };
+}
+
+/**
+ * Shows or hides the dialpad button.
+ *
+ * @param {boolean} show - Flag showing whether to show button or not.
+ * @returns {Function}
+ */
+export function showDialPadButton(show: boolean): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonName = 'dialpad';
+        const shouldShow = UIUtil.isButtonEnabled(buttonName) && show;
+
+        if (shouldShow) {
+            dispatch(setToolbarButton(buttonName, {
+                hidden: false
+            }));
+        }
+    };
+}
+
+/**
+ * Shows recording button.
+ *
+ * @returns {Function}
+ */
+export function showRecordingButton(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const eventEmitter = APP.UI.eventEmitter;
+        const buttonName = 'recording';
+
+        dispatch(setToolbarButton(buttonName, {
+            hidden: false
+        }));
+
+        Recording.init(eventEmitter, config.recordingType);
+    };
+}
+
+/**
+ * Shows or hides the 'shared video' button.
+ *
+ * @returns {Function}
+ */
+export function showSharedVideoButton(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonName = 'sharedvideo';
+        const shouldShow
+            = UIUtil.isButtonEnabled(buttonName)
+                && !config.disableThirdPartyRequests;
+
+        if (shouldShow) {
+            dispatch(setToolbarButton(buttonName, {
+                hidden: false
+            }));
+        }
+    };
+}
+
+/**
+ * Shows SIP call button if it's required and appropriate
+ * flag is passed.
+ *
+ * @param {boolean} show - Flag showing whether to show button or not.
+ * @returns {Function}
+ */
+export function showSIPCallButton(show: boolean): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonName = 'sip';
+        const shouldShow
+            = APP.conference.sipGatewayEnabled()
+                && UIUtil.isButtonEnabled(buttonName)
+                && show;
+
+        if (shouldShow) {
+            dispatch(setToolbarButton(buttonName, {
+                hidden: !shouldShow
+            }));
+        }
+    };
+}
+
+/**
+ * Shows the toolbar for specified timeout.
+ *
+ * @param {number} timeout - Timeout for showing the toolbar.
+ * @returns {Function}
+ */
+export function showToolbar(timeout: number = 0): Object {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        if (interfaceConfig.filmStripOnly) {
+            return;
+        }
+
+        const state = getState();
+        const { toolbarTimeout, visible } = state['features/toolbar'];
+        const finalTimeout = timeout ? timeout : toolbarTimeout;
```

Comment:
The above is not the simplest way to write the source code. Please pay attention and take care of such cases yourself in the future: `timeout ? timeout : toolbarTimeout` is equivalent to `timeout || toolbarTimeout`.